src/relay/transforms/dead_code.cc

```diff
@@ -82,7 +82,7 @@ class Eliminator : private ExprMutator {
 
   Expr VisitExpr_(const LetNode* op) final {
     Var v = op->var;
-    if (HasLet(v)) {
+    if (HasLet(v) || op->value.as<RefWriteNode>()) {
```

Comment:
I don't believe this fully fixes DCE due to nesting ref_write deep inside lets that get DCEd

Reply:
This should stop it from removing RefWrites anywhere and by extension keep all ref operations alive as the reference is now live too.

Reply:
I will give you a test case for this since I made this exact change when debugging and it wasn't enough to fix the problem. (In fact I remember segfaulting...)

Reply:
BTW, out of the scope of this PR, do we need to do some alias analysis to handle reference for DCE?

Reply:
I believe so, for now we should probably require DCE to error when the incoming module contains references. @jroesch and I are planning on writing an alias analysis pass soon.